### PR TITLE
Setup Prettier and Husky

### PR DESCRIPTION
**Pull request by nebula-aac**
**Notes for Reviewers**

This PR fixes 
- https://github.com/layer5io/sistent/issues/41

This PR setups prettier at the workspace level
Uses husky for `pre-commit`
Add `prettier-plugin-organize-imports` to help organize imports

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by nebula-aac** [comment]
will be adding lint-staged too

**Comment by leecalcote** [comment]
Merge conflicts...

File: .prettierrc [hunk 1]
```diff
@@ -1,8 +1,9 @@
 {
-    "semi": true,
-    "tabWidth": 2,
-    "printWidth": 100,
-    "singleQuote": true,
-    "trailingComma": "none",
-    "jsxBracketSameLine": true
+  "semi": true,
+  "tabWidth": 2,
```
Comment (leecalcote) [inline .prettierrc:3]: ✅

**Review by leecalcote — APPROVED** [review]

**Comment by nebula-aac** [comment]
Should be good now.

I fixed the `husky` to run as a `pre-commit` to format and lint-staged with eslint. Since there is no eslint rules yet, I will be adding this after this PR.

@leecalcote everything looks okay now?

**Review by leecalcote — APPROVED** [review]
LGTM

**Comment by leecalcote** [comment]
FYI - 

```
➜  sistent git:(configure-prettier) make
yarn run build-all
yarn run v1.22.19
$ yarn workspaces run build

> @layer5/sistent-components
warning package.json: No license field
$ vite build && tsc
/bin/sh: vite: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed.
Exit code: 127
Command: /usr/local/Cellar/node@18/18.17.1/bin/node
Arguments: /usr/local/Cellar/yarn/1.22.19/libexec/lib/cli.js run build
Directory: /Users/lee/code/sistent/packages/components
Output:

info Visit https://yarnpkg.com/en/docs/cli/workspaces for documentation about this command.
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [build] Error 127
```

**Comment by leecalcote** [comment]
I might be doing something wrong. I got many of these:

```
../../node_modules/@mui/base/useList/useList.js (1:0) Module level directives cause errors when bundled, "use client" in "../../node_modules/@mui/base/useList/useList.js" was ignored.
../../node_modules/@mui/base/useList/useListItem.js (1:0) Module level directives cause errors when bundled, "use client" in "../../node_modules/@mui/base/useList/useListItem.js" was ignored.
../../node_modules/@mui/base/useList/useListChangeNotifiers.js (1:0) Module level directives cause errors when bundled, "use client" in "../../node_modules/@mui/base/useList/useListChangeNotifiers.js" was ignored.
../../node_modules/@mui/base/utils/useLatest.js (1:0) Module level directives cause errors when bundled, "use client" in "../../node_modules/@mui/base/utils/useLatest.js" was ignored.
../../node_modules/@mui/base/utils/useTextNavigation.js (1:0) Module level directives cause errors when bundled, "use client" in "../../node_modules/@mui/base/utils/useTextNavigation.js" was ignored.
../../node_modules/@mui/base/utils/useForcedRerendering.js (1:0) Module level directives cause errors when bundled, "use client" in "../../node_modules/@mui/base/utils/useForcedRerendering.js" was ignored.
../../node_modules/@mui/base/utils/useMessageBus.js (1:0) Module level directives cause errors when bundled, "use client" in "../../node_modules/@mui/base/utils/useMessageBus.js" was ignored.
```
with this at the end:
```
✓ 1014 modules transformed.
dist/index.html                  0.39 kB │ gzip:  0.27 kB
dist/assets/index-f7b78167.js  214.71 kB │ gzip: 71.71 kB
✓ built in 2.99s
```

**Pull request merged**